picard/webservice.py

```diff
@@ -134,6 +135,87 @@ def _read_xml(stream):
     return document
 
 
+class WsRequest(object):
+
+    def __init__(self, method, host, port, path, data, handler, xml,
+                 mblogin=False, cacheloadcontrol=None, refresh=None,
+                 access_token=None, queryargs=None, retries=0,
+                 priority=False, important=False):
+        self.method = method
+        self.host = host
+        self.port = port
+        self.path = path
+        self.data = data
+        self.handler = handler
+        self.xml = xml
+        self.mblogin = mblogin
+        self.cacheloadcontrol = cacheloadcontrol
+        self.refresh = refresh
+        self.access_token = access_token
+        self.queryargs = queryargs
+        self.priority = priority
+        self.important = important
+        self.retries = retries
+
+        self._qrequest = None
```

Comment:
Unless I'm mistaken, you can just call `build_qurl` here instead of having a property for `qrequest`.